Composer/packages/client/src/pages/knowledge-base/QnAPage.tsx

```diff
@@ -21,17 +21,18 @@ import TableView from './table-view';
 
 const CodeEditor = React.lazy(() => import('./code-editor'));
 
-interface QnAPageProps extends RouteComponentProps<{}> {
-  projectId?: string;
+const QnAPage: React.FC<RouteComponentProps<{
```

Comment:
RouteComponentProps<TParams> makes the passed TParams partial, so all of them will be optional.